src/main/java/uk/gov/companieshouse/api/accounts/service/impl/AbstractServiceImpl.java

```diff
@@ -0,0 +1,41 @@
+package uk.gov.companieshouse.api.accounts.service.impl;
+
+import org.springframework.data.mongodb.repository.MongoRepository;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.GenerateEtagUtil;
+import uk.gov.companieshouse.api.accounts.model.entity.BaseEntity;
+import uk.gov.companieshouse.api.accounts.model.rest.RestObject;
+import uk.gov.companieshouse.api.accounts.service.AbstractService;
+import uk.gov.companieshouse.api.accounts.transformer.GenericTransformer;
+
+@Service
+public abstract class AbstractServiceImpl<C extends RestObject, E extends BaseEntity> implements
+        AbstractService<C, E> {
+
+    private MongoRepository mongoRepository;
+
+    private GenericTransformer<C, E> genericTransformer;
+
+    public AbstractServiceImpl(MongoRepository mongoRepository,
+            GenericTransformer<C, E> genericTransformer) {
+        this.mongoRepository = mongoRepository;
+        this.genericTransformer = genericTransformer;
+    }
+
+    @Override
+    public C save(C rest) {
```

Comment:
What does `C` stand for? It doesn't seem to follow the standard generic type parameter conventions?

Reply:
Suggest different letters and I will change them

Reply:
It's not that you should change them, I'm just trying to gauge what C represents?